docs/query-builder.md

```diff
@@ -448,64 +448,136 @@ DB::collection('items')
+
 ```php
 DB::collection('users')
-    ->where('name', 'John')
-    ->push('messages', [
-        'from' => 'Jane Doe',
-        'message' => 'Hi John',
-    ]);
+  ->where('name', 'John')
+    // Push an embedded document as a value to the messages array
+  ->push('messages', [
+      [ 'from'    => 'Jane Doe', 'message' => 'Hi John' ]
+  ]);
+// Result:
+// messages: [
+//      { from: "Jane Doe", message: "Hi John" }
+//  ]
+
+// Or
 
 $user->push('messages', [
-    'from' => 'Jane Doe',
-    'message' => 'Hi John',
+    [ 'from'    => 'Jane Doe', 'message' => 'Hi John' ]
 ]);
 ```
 
-If you **DON'T** want duplicate items, set the third parameter to `true`:
+If you **DON'T** want duplicate values, set the third parameter to `true`:
 
 ```php
 DB::collection('users')
-    ->where('name', 'John')
-    ->push('items', 'boots', true);
+  ->where('name', 'John')
+  ->push('items', 'boots');
+// Result:
+// items: ['boots']
+
+DB::collection('users')
+  ->where('name', 'John')
+  ->push('items', ['hat', 'boots', 'jeans'], true);
+// Result:
+// items: ['boots', 'hat', 'jeans']
+
+// Or
 
-$user->push('items', 'boots', true);
+$user->push('messages', [
+    [ 'from'    => 'Jane Doe', 'message' => 'Hi John' ]
+]);
+// Result:
+// messages: [
+//      { from: "Jane Doe", message: "Hi John" }
+//  ]
+
+$user->push('messages', [
+    [ 'from'    => 'Jess Doe', 'message' => 'Hi' ],
+    [ 'from'    => 'Jane Doe', 'message' => 'Hi John' ],
+], true);
+// Result:
+// messages: [
+//      { from: "Jane Doe", message: "Hi John" }
+//      { from: "Jess Doe", message: "Hi" }
+//  ]
 ```
 
 **Pull**
 
-Remove an item from an array.
+Remove one or multiple values from the `items` array.
 
 ```php
+// items: ['boots', 'hat', 'jeans']
+
 DB::collection('users')
-    ->where('name', 'John')
-    ->pull('items', 'boots');
+  ->where('name', 'John')
+  ->pull('items', 'boots'); // Pull a single value
+// Result:
+// items: ['hat', 'jeans']
 
-$user->pull('items', 'boots');
+// Or pull multiple values
+
+$user->pull('items', ['boots', 'jeans']);
+// Result:
+// items: ['hat']
 ```
 
+Embedded document and arrays values can also be pulled from the `messages` array.
+
 ```php
+// Latest state:
+// messages: [
+//      { from: "Jane Doe", message: "Hi John" }
+//      { from: "Jess Doe", message: "Hi" }
+//  ]
+
 DB::collection('users')
-    ->where('name', 'John')
-    ->pull('messages', [
-        'from' => 'Jane Doe',
-        'message' => 'Hi John',
-    ]);
+  ->where('name', 'John')
+    // It will search for the exact array and pulls it
+  ->pull('messages', [
+      [ 'from' => 'Jane Doe', 'message' => 'Hi John' ]
+  ]);
+// Result:
+// messages: [
+//      { from: "Jess Doe", message: "Hi" }
+//  ]
+
+// Or pull multiple array
 
 $user->pull('messages', [
-    'from' => 'Jane Doe',
-    'message' => 'Hi John',
+    [ 'from' => 'Jane Doe', 'message' => 'Hi John' ],
+    [ 'from' => 'Jess Doe', 'message' => 'Hi' ]
```

Comment:
Again, the field order shouldn't matter here if the library is always using `$pull`.

Reply:
I see. Thanks.